request_extractor.go

```diff
@@ -32,19 +33,55 @@ func extractRequestInfo(ctx context.Context) (*RequestJSON, *http.Request) {
 // understands from the given HTTP request. Returns the sub-object supported by
 // the notify API.
 func extractRequestInfoFromReq(req *http.Request) *RequestJSON {
-	proto := "http://"
-	if req.TLS != nil {
-		proto = "https://"
-	}
 	return &RequestJSON{
 		ClientIP:   req.RemoteAddr,
 		HTTPMethod: req.Method,
-		URL:        proto + req.Host + req.RequestURI,
+		URL:        sanitizeURL(req),
 		Referer:    req.Referer(),
 		Headers:    parseRequestHeaders(req.Header),
 	}
 }
 
+// sanitizeURL will build up the URL matching the request. It will filter query parameters to remove sensitive fields.
+// The query part of the URL might appear differently (different order of parameters) if any filtering was done.
+func sanitizeURL(req *http.Request) string {
+	scheme := "http"
+	if req.TLS != nil {
+		scheme = "https"
+	}
+
+	rawQuery := req.URL.RawQuery
+	parsedQuery, err := url.ParseQuery(req.URL.RawQuery)
+	if err == nil {
+		changed := false
+		for key, values := range parsedQuery {
+			if contains(Config.ParamsFilters, key) {
+				for i, v := range values {
+					if len(v) == 0 {
+						// No need to filter empty parameters.
+						continue
+					}
+
+					values[i] = "FILTERED"
```

Comment:
Good point about `[]` being represented differently in a URL, I didn't think of that. I think we can get around this though by doing a string replacement after we've rendered the URL.
Just in case `"FILTERED"` is ever used as a (valid) URL param value, maybe we should do
```suggestion
					values[i] = "BUGSNAG_URL_FILTERED"
```
here, and then replace `"BUGSNAG_URL_FILTERED"` with `"[FILTERED]"` just before returning from this function.